pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/OpChain.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.runtime.operator;
+
+import com.google.common.base.Suppliers;
+import java.util.List;
+import java.util.function.Supplier;
+import org.apache.pinot.common.request.context.ThreadTimer;
+import org.apache.pinot.core.common.Operator;
+import org.apache.pinot.query.mailbox.MailboxIdentifier;
+import org.apache.pinot.query.runtime.blocks.TransferableBlock;
+
+
+/**
+ * An {@code OpChain} represents a chain of operators that are separated
+ * by send/receive stages.
+ */
+public class OpChain {
+
+  private final Operator<TransferableBlock> _root;
+  private final List<MailboxIdentifier> _inputMailboxes;
+  // TODO: build timers that are partial-execution aware
+  private final Supplier<ThreadTimer> _timer;
+
+  public OpChain(Operator<TransferableBlock> root, List<MailboxIdentifier> inputMailboxes) {
+    _root = root;
+    _inputMailboxes = inputMailboxes;
+
+    // use memoized supplier so that the timing doesn't start until the first time we get
+    // the timer
+    _timer = Suppliers.memoize(ThreadTimer::new)::get;
+  }
+
+  public Operator<TransferableBlock> getRoot() {
+    return _root;
+  }
+
+  public List<MailboxIdentifier> getInputMailboxes() {
+    return _inputMailboxes;
+  }
```

Comment:
doesn't seem like this is being used. from what I understand on the triggering mechanism:
- register
- onDataAvailable
- next().getRoot().nextBlock() completes

all should be able to identify which opChain to call based on jobID alone. do we need the input mailbox id list?

Reply:
I think I see what you're saying. Since this isn't used in this PR I'll just clean it up for now and pipe it back in when I add the PR which triggers the scheduler via mailbox data available.

Reply:
thank you. or adding a unit-test to explain is also good. i am assuming it will be related to https://github.com/apache/pinot/pull/9753#discussion_r1023096179 but I am not sure exactly. so i pop the question.